environmentManager/index.js

```diff
@@ -3,78 +3,109 @@ const { validEnvironment, validEnvironmentConfig } = require('../utils/parserCon
-      settings.core.authorizationKey = apiKey;
+        validEnvironment(environment);
+        const authToken = environment['AUTH_TOKEN'];
+        const apiKey = environment['API_KEY'];
+        settings.core.authorizationKey = apiKey;
 
-      this._environments[authToken] = {
-        apiKey: apiKey,
-        factory: getSplitFactory(settings),
-        isClientReady: false,
-      };
+        this._environments[authToken] = {
+          apiKey: apiKey,
+          factory: getSplitFactory(settings),
+          isClientReady: false,
+        };
 
-      this._clientReadiness(this._environments[authToken].factory.client());
+        const client = this._environments[authToken].factory.client();
+        this._clientReadiness(client);
 
-    });
-    this.ready();
-  }
 
-  _clientReadiness(client){
-    this._readyPromises.push(client.ready());
-    client.on(client.Event.SDK_READY, () => client.isClientReady = true);
-  }
+      });
+      this.ready();
+    }
 
-  getFactory(authToken) {
-    return this._environments[authToken].factory;
-  }
+    _clientReadiness(client){
+      this._readyPromises.push(client.ready());
+      client.on(client.Event.SDK_READY, () => client.isClientReady = true);
+    }
 
-  getVersion() {
-    return this._environments[this.getAuthTokens()[0]].factory.settings.sdkVersion;
-  }
+    getFactory(authToken) {
+      return this._environments[authToken].factory;
+    }
 
-  getClient(authToken) {
-    return this._environments[authToken].factory.client();
-  }
+    getVersion() {
+      return this._environments[this.getAuthTokens()[0]].factory.settings.sdkVersion;
+    }
 
-  getManager(authToken) {
-    return this._environments[authToken].factory.manager();
-  }
+    getClient(authToken) {
+      return this._environments[authToken].factory.client();
+    }
 
-  validToken(authToken) {
-    return Object.keys(this._environments).indexOf(authToken) > -1;
-  }
+    getManager(authToken) {
+      return this._environments[authToken].factory.manager();
+    }
 
-  getAuthTokens() {
-    return Object.keys(this._environments);
-  }
+    validToken(authToken) {
+      return Object.keys(this._environments).indexOf(authToken) > -1;
+    }
 
-  ready() {
-    return Promise.allSettled(this._readyPromises).then(this._clientsReady = true);
-  }
+    getAuthTokens() {
+      return Object.keys(this._environments);
+    }
+
+    async ready() {
+      await Promise.all(this._readyPromises).then(this._clientsReady = true);
+    }
+
+    async destroy() {
+      this.getAuthTokens().forEach(async authToken => {
+        const client = this.getClient(authToken);
+        await client.destroy();
+      });
+      this._clientsReady = false;
+    }
 
-  isReady() {
-    return this._clientsReady;
+    isReady() {
+      return this._clientsReady;
+    }
   }
-}
 
-module.exports = new EnvironmentManager();
\ No newline at end of file
+  let instance;
+
+  return {
+    hasInstance: function() {
```

Comment:
Just a style suggestion, you can replace it with `hasInstance() {` for a shorter code. 
Check other method definitions too.